src/test/results.rs

```diff
@@ -79,67 +80,88 @@ impl From<&Test> for Results {
             test.words.iter().flat_map(|w| w.events.iter()).collect();
 
         Self {
-            timing: {
-                let mut timing = TimingData {
-                    overall_cps: -1.0,
-                    per_event: Vec::new(),
-                    per_key: HashMap::new(),
-                };
-
-                // map of keys to a two-tuple (total time, clicks) for counting average
-                let mut keys: HashMap<KeyEvent, (f64, usize)> = HashMap::new();
-
-                for win in events.windows(2) {
-                    let event_dur = win[1]
-                        .time
-                        .checked_duration_since(win[0].time)
-                        .map(|d| d.as_secs_f64());
-
-                    if let Some(event_dur) = event_dur {
-                        timing.per_event.push(event_dur);
-
-                        let key = keys.entry(win[1].key).or_insert((0.0, 0));
-                        key.0 += event_dur;
-                        key.1 += 1;
-                    }
-                }
+            timing: Self::calc_timing(&events),
+            accuracy: Self::calc_accuracy(&events),
+            missed_words: Self::calc_missed_words(&test),
+        }
+    }
+
+}
 
-                timing.per_key = keys
-                    .into_iter()
-                    .map(|(key, (total, count))| (key, total / count as f64))
-                    .collect();
-
-                timing.overall_cps =
-                    timing.per_event.len() as f64 / timing.per_event.iter().sum::<f64>();
-
-                timing
-            },
-            accuracy: {
-                let mut acc = AccuracyData {
-                    overall: Fraction::new(0, 0),
-                    per_key: HashMap::new(),
-                };
-
-                events
-                    .iter()
-                    .filter(|event| event.correct.is_some())
-                    .for_each(|event| {
-                        let key = acc
-                            .per_key
-                            .entry(event.key)
-                            .or_insert_with(|| Fraction::new(0, 0));
-
-                        acc.overall.denominator += 1;
-                        key.denominator += 1;
-
-                        if event.correct.unwrap() {
-                            acc.overall.numerator += 1;
-                            key.numerator += 1;
-                        }
-                    });
-
-                acc
-            },
+impl Results {
```

Comment:
I'm fine with extracting the computation of each property out to its own function, but they should be free rather than in a `Result` impl.

Reply:
That makes sense. I was kind of split on this anyway. will make the update.

Reply:
I made this update